### PR TITLE
[DOCS] Writing a new Arduino documentation about GPIO

### PR DESCRIPTION
**Pull request by halyssonJr**
----------------------------------------------------------------------------------------------------------------------------------------------------
### Checklist

1. [ ] Created five sections (About, GPIO Configuration, API, Simulation and Exemple Code)
2. [ ] Created simulation application 
----------------------------------------------------------------------------------------------------------------------------------------------------

## Summary
Arduino Documentation for GPIO 

## Impact
None

Co-authored-by: Pedro Minatel <pminatel@gmail.com>

### CODE REVIEW

**Comment by pedrominatel** [comment]
@halyssonJr first things first, remove the build folder. I'll check why the `.gitignore` is still considering the `_build` folder when adding the files.

File: docs/source/api/gpio.rst [hunk 1]
```diff
@@ -1,3 +1,115 @@
...
+
+pinMode
+***********
+
+Initial configuration for use pin with input or output.  
+
+
+.. code-block:: arduino
+
+    void piMode(uint8_t pin, uint8_t mode);
```
Comment (atanisoft) [inline docs/source/api/gpio.rst:43]: typo: piMode -> pinMode

File: docs/source/api/gpio.rst [hunk 2]
```diff
@@ -1,3 +1,115 @@
...
+
+.. code-block:: arduino
+
+    void digitalWrite(uint8_t pin, uint8_t val);
+
+* ``pin``  select a GPIO
+* ``val``  logic level write on pin (``HIGH`` or ``LOW``) 
+
+.. warning:: 
+  If pin not set to ``OUTPUT`` on  pinMode(), may do not work correct.
```
Comment (atanisoft) [inline docs/source/api/gpio.rst:61]: Remove "do" from the second half of the sentence

File: docs/source/api/gpio.rst (hunk 1 above)
Comment (halyssonJr) [reply docs/source/api/gpio.rst:43]: Sorry about that, i will be fix

**Comment by pedrominatel** [comment]
@halyssonJr please be aware of not adding the unrelated folders and files of your build and environment to the commit. Add only the files that you are working on.

Can you clarify why you've changed the `docs/source/conf.py` file?

**Comment by halyssonJr** [comment]
> @pedrominatel, maybe i just have uncommented, but not made any changes directly  to the `docs/source/conf.py` file?

File: docs/source/api/gpio.rst [hunk 3]
```diff
@@ -1,3 +1,115 @@
 ####
 GPIO
 ####
+About
+-----
+
+GPIO (General Purpouse Input Output) is peripheral responsible to controls reception and 
+deliver digital signals by means of easy interface acess with world, physical pins.
+
+A fetures of this peripheral relates with largely quantity pins available on chip, also being
```
Comment (VojtechBartoska) [inline docs/source/api/gpio.rst:10]: typo: fe**a**tures

**Review by VojtechBartoska — DISMISSED** [review]

**Review by pedrominatel — CHANGES_REQUESTED** [review]
PTAL at my considerations.

File: docs/source/api/gpio.rst [hunk 4]
```diff
@@ -1,3 +1,115 @@
...
+    int digitalRead(uint8_t pin);
+
+* ``pin`` select GPIO
+
+This function will be return logical state ``HIGH`` or ``LOW``
+
+Simulation
+----------
+
+This application helps understand concepts using dynamic simulation provide by `Wokwi`_, test the gpio code and start.
```
[the remark this answers is not in the document]
Comment (halyssonJr) [reply docs/source/api/gpio.rst:79]: Ok, thanks

**Comment by halyssonJr** [comment]
@pedrominatelI,  I try to split my commit. So I can't upload just the files I changed. In the next PR I will fix these problems

**Comment by pedrominatel** [comment]
@halyssonJr your PR is messed. You were supposed to edit only one file but now you have 191!

If you need help to understand the Git flow I'll be glad to help. Currently, your PR is not following the expectations and cant be reviewed. Ping me later if you need help.

**Comment by pedrominatel** [comment]
@halyssonJr please take a look at my modifications.

File: docs/source/api/gpio.rst [hunk 5]
```diff
@@ -1,3 +1,170 @@
...
+
+To read the state of a given pin configured as ``INPUT``, the function ``digitalRead`` is used.
+
+.. code-block:: arduino
+
+    int digitalRead(uint8_t pin);
+
+* ``pin`` select GPIO
+
+This function will return the logical state of the selecetd pin as ``HIGH`` or ``LOW``.
```
Comment (VojtechBartoska) [inline docs/source/api/gpio.rst:80]: typo: select**e**d

**Review by VojtechBartoska — APPROVED** [review]
Overall looks good. Found only one small typo.

**Pull request merged**